### PR TITLE
statistics: drop old stats meta after truncate a table 

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/49663

Problem Summary:

### What changed and how does it work?

- Added a test case to cover this problem.
- Drop the old stats meta after truncating a table.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix TiDB can not drop old stats meta after truncating a table issue
```

### CODE REVIEW

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check

**Review by hawkingrei — APPROVED** [review]

File: pkg/statistics/handle/ddl/ddl.go [hunk 1]
```diff
@@ -71,6 +71,11 @@ func (h *ddlHandlerImpl) HandleDDLEvent(t *util.DDLEvent) error {
 				return err
 			}
 		}
+
+		// Remove the old table stats.
+		if err := h.statsWriter.ResetTableStats2KVForDrop(droppedTableInfo.ID); err != nil {
```
Comment (hi-rustin) [inline pkg/statistics/handle/ddl/ddl.go:76]: We should drop all partitions as well.

**Review by qw4990 — APPROVED** [review]

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check

**Pull request merged**